searchcore/src/vespa/searchcore/proton/server/storeonlyfeedview.cpp

```diff
@@ -632,8 +633,10 @@ StoreOnlyFeedView::adjustMetaStore(const DocumentOperation &op, const GlobalId &
                 putMetaData(_metaStore, docId, op, _params._subDbType == SubDbType::REMOVED);
             }
         } else if (op.getValidPrevDbdId(_params._subDbId)) {
-            _gidToLidChangeHandler.notifyRemove(gid, serialNum);
+            vespalib::Gate gate;
```

Comment:
Consider using std::promise&lt;void&gt;, std::future&lt;void&gt;, and a context that just sets the promise in the destructor.

Reply:
I think it is clearer to use the Gate than the general promise/future trick.